content/chapters/io/lecture/demo/devices/hwaddr-ioctl.c

```diff
@@ -16,31 +16,36 @@
 
 #include "utils/utils.h"
 
-#define IFNAME			"enp0s31f6"
-
-int main(void)
+int main(int argc, const char *argv[])
 {
+	if (argc < 2) {
+		printf("Usage: %s interface_name\n", argv[0]);
+		return 0;
+	}
 	int s;
 	struct ifreq ifr;
 	int rc;
 	size_t i;
+	char *ifname = malloc(strlen(argv[1]));
+	strcpy(ifname, argv[1]);
 
 	// Create socket
 	s = socket(PF_INET, SOCK_DGRAM, 0);
 	DIE(s < 0, "socket");
 
 	// Call ioctl to populate ifr struct
-	snprintf(ifr.ifr_name, strlen(IFNAME)+1, "%s", IFNAME);
+	snprintf(ifr.ifr_name, strlen(ifname)+1, "%s", ifname);
```

Comment:
Ditto. Use `argv[1]` instead of `ifname`.

Reply:
Done.